GenderPayGap.WebJob/Functions/Functions.Purge.cs

```diff
@@ -27,48 +28,71 @@ public async Task PurgeUsers([TimerTrigger("40 3 * * *" /* 03:40 once per day */
+            
+            foreach (User user in users)
+            {
+                PurgeUser(user, pinExpiryDate, runId, startTime);
+            }
+
+            LogFunctionEnd(runId, nameof(PurgeUsers), startTime);
+        }
+
+        private static async void PurgeUser(User user, DateTime pinExpiryDate, string runId, DateTime startTime)
+        {
             try
             {
-                DateTime deadline = VirtualDateTime.Now.AddDays(0 - Global.PurgeUnverifiedUserDays);
-                List<User> users = await _DataRepository.GetAll<User>()
-                    .Where(u => u.EmailVerifiedDate == null && (u.EmailVerifySendDate == null || u.EmailVerifySendDate.Value < deadline))
-                    .ToListAsync();
-                DateTime pinExpireyDate = VirtualDateTime.Now.AddDays(0 - Global.PinInPostExpiryDays);
-                foreach (User user in users)
+                //Ignore if they have verified PIN
+                if (user.UserOrganisations.Any(
+                    uo => uo.PINConfirmedDate != null || (uo.PINSentDate != null && uo.PINSentDate < pinExpiryDate)))
                 {
-                    //Ignore if they have verified PIN
-                    if (user.UserOrganisations.Any(
-                        uo => uo.PINConfirmedDate != null || uo.PINSentDate != null && uo.PINSentDate < pinExpireyDate))
-                    {
-                        continue;
-                    }
-
-                    var logItem = new ManualChangeLogModel(
-                        nameof(PurgeUsers),
-                        ManualActions.Delete,
-                        AppDomain.CurrentDomain.FriendlyName,
-                        nameof(user.UserId),
-                        user.UserId.ToString(),
-                        null,
-                        JsonConvert.SerializeObject(new {user.UserId, user.EmailAddress, user.JobTitle, user.Fullname}),
-                        null);
-                    _DataRepository.Delete(user);
-                    await _DataRepository.SaveChangesAsync();
-                    await Global.ManualChangeLog.WriteAsync(logItem);
+                    return;
                 }
 
-                LogFunctionEnd(runId, nameof(PurgeUsers), startTime);
+                var logItem = new ManualChangeLogModel(
+                    nameof(PurgeUsers),
+                    ManualActions.Delete,
+                    AppDomain.CurrentDomain.FriendlyName,
+                    nameof(user.UserId),
+                    user.UserId.ToString(),
+                    null,
+                    JsonConvert.SerializeObject(new {user.UserId, user.EmailAddress, user.JobTitle, user.Fullname}),
+                    null);
+            
+                DeleteUserAndAuditLogs(user);
+                await Global.ManualChangeLog.WriteAsync(logItem);
             }
             catch (Exception ex)
             {
                 LogFunctionError(runId, nameof(PurgeUsers), startTime, ex );

```

Comment:
This should get picked up by App Insights and the new alert, which means we no longer need to send an email on line 66